### PR TITLE
Various fixes for getattr/setattr/hasattr

### PR DESCRIPTION
**Pull request by skinny121**
Fixes the following:
- setattr() now calls \_\_setattr\_\_ and implemented object.\_\_setattr\_\_
- Fix \_\_getattr\_\_ getting called with the class instead of the instance
- Fix hasattr catching non-AttributeError exceptions
- Added default paramter to getattr
- object() had a dict
- property not returning itself for class bindings

### CODE REVIEW

**Comment by skinny121** [comment]
No, that's fine, I will rebase to remove the formatting commit.

File: vm/src/obj/objnone.rs [hunk 1]
```diff
@@ -39,6 +42,57 @@ impl PyNoneRef {
     fn bool(self, _vm: &mut VirtualMachine) -> PyResult<bool> {
         Ok(false)
     }
+
+    fn get_attribute(self, name: PyStringRef, vm: &mut VirtualMachine) -> PyResult {
```
Comment (windelbouwman) [inline vm/src/obj/objnone.rs]: This piece I do not get. Do we want to remove it in future again?

**Comment by windelbouwman** [comment]
Btw, I like the fact that on the virtualmachine we have now most methods available, such as setattr, getattr, delattr and friends. They simply invoke the corresponding magic method.

File: vm/src/obj/objnone.rs (hunk 1 above)
Comment (skinny121) [reply vm/src/obj/objnone.rs]: What do you mean again? We may be able to remove this in the future once we have slots implemented for the core operations. As we currently don't, we have no way of representing Option<PyObjectRef> as a PyValue, without conflating Some(PyNone) with None. So I had to find a workaround for cases where it mattered, which are properties on None.

**Comment by windelbouwman** [comment]
@skinny121 I want to merge this, but there is still a conflicting file. Could you have a look at this?

**Comment by skinny121** [comment]
@windelbouwman I have resolved the conflict and removed the formatting commit by performing formatting on every commit.

**Pull request merged**